aiohttp_jinja2/__init__.py

```diff
@@ -43,12 +39,36 @@
     _SimpleTemplateHandler, _MethodTemplateHandler, _ViewTemplateHandler
 ]
 
+_ContextProcessor = Callable[[web.Request], Awaitable[Dict[str, Any]]]
+
+if sys.version_info >= (3, 8):
+    from typing import Protocol
+
+    class _TemplateWrapped(Protocol):
+        @overload
+        async def __call__(self, request: web.Request) -> web.StreamResponse:
+            ...
+
+        @overload
+        async def __call__(self, view: AbstractView) -> web.StreamResponse:
+            ...
+
+        @overload
+        async def __call__(
+            self, _self: Any, request: web.Request
```

Comment:
What is the overload for? Middleware?

Reply:
There's 3 different signatures that are valid for the wrapper to work:
```
@template
async def foo(request)  # first overload

class V(web.View):
    @template
    def foo(self):  # second overload

class Foo:
    @template
    def foo(self, request)  # third overload
```

Reply:
Got you